lib/ansible/modules/net_tools/netplan.py

```diff
@@ -0,0 +1,903 @@
+    description:
+      - The YAML currently being used
+    required: false
+    type: int
+
+  interface-id:
+    description:
+      - Interface ID that must be unique.
+    required: true
+
+  type:
+    description:
+      - Interface type. Depends on type, you can use specific params
+        (E.g bridges can use stp param).
+    choices: [ bridges, bonds, ethernets, vlans, wifis ]
+    required: true
+
+  state:
+    description:
+      - C(present) define a net interface on C(filename);
+        C(absent) undefine a net interface on C(filename).
+    choices: [ present, absent ]
+    required: true
+
+  dhcp4:
+    description:
+      - Enable/Disable DHCP for IPv4. Default is disabled.
+    required: false
+    type: bool
+
+  dhcp6:
+    description:
+      - Enable/Disable DHCP for IPv6. Default is disabled.
+    required: false
+    type: bool
+
+  link-local:
+    description:
+      - Configure the link-local addresses to bring up. Valid options are
+        ipv4 and 'ipv6', which respectively allow enabling IPv4 and IPv6
+        link local addressing.
+    choices: [ ipv4, ipv6 ]
+    required: false
+
+  critical:
+    description:
+      - Designate the connection as "critical to the system", meaning that
+        special care will be taken by systemd-networkd to not release the IP
+        from DHCP when it the daemon is restarted. Networkd backend only.
+    required: false
+    type: bool
+
+  dhcp-identifier:
+    description:
+      - When set to C(mac); pass that setting over to systemd-networkd to use
+        the device's MAC address as a unique identifier rather than a
+        RFC4361-compliant Client ID. This has no effect when NetworkManager
+        is used as a renderer.
+    required: false
+
+  accept-ra:
+    description:
+      - Accept Router Advertisement that would have the kernel configure IPv6
+        by itself. When enabled, accept Router Advertisements. When disabled,
+        do not respond to Router Advertisements. If unset use the host kernel
+        default setting.
+    required: false
+    type: bool
+
+  addresses:
+    description:
+      - Add static addresses to the interface in addition to the ones received
+        through DHCP or RA. Each sequence entry is in CIDR notation, i. e. of
+        the form C(addr)/C(prefixlen). C(addr) is an IPv4 or IPv6 address as
+        recognized by C(inet_pton) and C(prefixlen) the number of bits of the
+        subnet.
+    required: false
+    type: list
+
+  gateway4:
+    description:
+      - Set default gateway for IPv4, for manual address configuration.
+        This requires setting addresses too. Gateway IPs must be in a form
+        recognized by C(inet_pton).
+    required: false
+
+  gateway6:
+    description:
+      - Set default gateway for IPv6, for manual address configuration.
+        This requires setting addresses too. Gateway IPs must be in a form
+        recognized by C(inet_pton).
+    required: false
+
+  nameservers-search:
```

Comment:
The name of that parameter is bad, it's simply “DNS search domains”.
Perhaps `search-domains`, or `resolvconf-domains` (but then rename `nameserver-addresses` to `resolvconf-addresses`)